File: QytetetJava/src/modeloqytetet/Qytetet.java

package modeloqytetet;

import java.util.ArrayList;
import java.util.Collections;

/**
 *
 * @author albertorodriguez
 */
public class Qytetet {
    
    //Singleton
    private static final Qytetet instance = new Qytetet();
    
    public static int MAX_JUGADORES = 4;
    static final int MAX_CARTAS = 10;
    static final int MAX_CASILLAS = 20;
    static final int PRECIO_LIBERTAD = 200;
    static final int SALDO_SALIDA = 1000;
    
    //Atributos de la clase Sorpresa
    private Sorpresa cartaActual=null; 
    private ArrayList<Sorpresa> mazo = new ArrayList();
    
    //Atributo de la clase Tablero
    private Tablero tablero = new Tablero();
    
    //Atributos de la clase Jugador
    private Jugador jugadorActual;
    private ArrayList<Jugador> jugadores = new ArrayList(); 
    
    //Objeto de la clase DADO - Como es singleton solo se puede crear un Objeto.
    Dado dado = Dado.getInstance();
        
    //CONSTRUCTOR PRIVADO PARA ASEGURAR QUE NO SE PUEDE INSTANCIAR
    private Qytetet(){}
    
    public static Qytetet getInstance(){
        instance.inicializarCartasSorpresa();
        instance.inicializarTablero();
        //instance.inicializarJugadores(nombres);
        return instance;
    }
        
    //--------------------------------------------------------------------------------------
    //-----------------------------MODIFICADORES/CONSULTORES--------------------------------

    public static int getMAX_JUGADORES() {
        return MAX_JUGADORES;
    }

    public static int getMAX_CARTAS() {
        return MAX_CARTAS;
    }

    public static int getMAX_CASILLAS() {
        return MAX_CASILLAS;
    }

    public static int getPRECIO_LIBERTAD() {
        return PRECIO_LIBERTAD;
    }

    public static int getSALDO_SALIDA() {
        return SALDO_SALIDA;
    }

    public ArrayList<Sorpresa> getMazo() {
        return mazo;
    }

    public Tablero getTablero() {
        return tablero;
    }

    public ArrayList<Jugador> getJugadores() {
        return jugadores;
    }

    public Dado getDado() {
        return dado;
    }
    
    public Sorpresa getCartaActual(){
        return cartaActual;
    }
    
      
    public Jugador getJugadorActual(){
        return jugadorActual;
    }
    
    
    
    
    //---------------------------------METODOS-------------------------------------
    
    public boolean aplicarSorpresa(){
        throw new UnsupportedOperationException("Sin Implementar"); 
    }
    
    public boolean cancelarHipoteca(Casilla casilla){
        throw new UnsupportedOperationException("Sin Implementar"); 
    }
    
    public boolean comprarTituloPropiedad(){
        throw new UnsupportedOperationException("Sin Implementar");
    }
    
    public boolean edificarCasa(Casilla casilla){
        throw new UnsupportedOperationException("Sin Implementar");
    }
    
    public boolean edificarHotel(Casilla casilla){
        throw new UnsupportedOperationException("Sin Implementar");
    }
     
    public boolean hipotecarPropiedad(Casilla casilla){
        throw new UnsupportedOperationException("Sin Implementar");
    }
    
    public void inicializarJuego(ArrayList<String> nombres){ 
        throw new UnsupportedOperationException("Sin Implementar");
    }
    
    public boolean intentarSalirCarcel(MetodoSalirCarcel metodo){ 
        throw new UnsupportedOperationException("Sin Implementar");
    }
    
    public boolean jugar(){
        throw new UnsupportedOperationException("Sin Implementar");
    }
    
    public ArrayList obtenerRanking(){ 
        ArrayList<Jugador> List = new ArrayList();
        
        return List;
    }
    
    public ArrayList propiedadesHipotecadasJugador(boolean hipotecadas){
        //ArrayList<Casilla> casillas = new ArrayList();
        throw new UnsupportedOperationException("Sin Implementar");
        //return casillas;
    }
    
    public Jugador siguienteJugador(){
        throw new UnsupportedOperationException("Sin Implementar");
    }
    
    public boolean venderPropiedad(Casilla casilla){
        throw new UnsupportedOperationException("Sin Implementar");
    }
    
    private void encarcelarJugador(){
        throw new UnsupportedOperationException("Sin Implementar");
    }
    
    private void inicializarCartasSorpresa(){
        mazo.add(new Sorpresa("¡Felicidades! Hoy es dia de pago, recibes: ", 100, TipoSorpresa.PAGARCOBRAR));
        mazo.add(new Sorpresa("¡Mala Suerte! Dia de impuestos, te toca pagar: ", 50, TipoSorpresa.PAGARCOBRAR));
        mazo.add(new Sorpresa("¡Mala Suerte! Directo a la carcel", tablero.getCarcel().getNumeroCasilla(), TipoSorpresa.IRACASILLA)); //Casilla 5 CARCEL);
        mazo.add(new Sorpresa("El mazo de cartas te manda a la casilla numero ", 12, TipoSorpresa.IRACASILLA));
        mazo.add(new Sorpresa("El mazo de cartas te manda a la casilla numero ", 19, TipoSorpresa.IRACASILLA));
        mazo.add(new Sorpresa("Hacienda te cobra", 200, TipoSorpresa.PORCASAHOTEL));
        mazo.add(new Sorpresa("Hacienda te devuelve", 50, TipoSorpresa.PORCASAHOTEL));
        mazo.add(new Sorpresa("¡Felicidades!, es tu cumpleaños todos los jugadores te pagan ", 50, TipoSorpresa.PORJUGADOR));
        mazo.add(new Sorpresa("Es tu dia de mala suerte, una apuesta te hace pagar a todos los jugadores", 50, TipoSorpresa.PORJUGADOR));
        mazo.add(new Sorpresa("¡Enhorabuena!, quedas en libertad ", 0, TipoSorpresa.PAGARCOBRAR));
   
        Collections.shuffle(mazo);
    }
   
    private void inicializarJugadores(ArrayList<String> nombres){ 
//       for(int i=0; i<getMAX_JUGADORES(); i++){
//        jugadores.add(new Jugador(nombres.get(i)));
//       }
       int i=0;
       while(nombres.get(i)!=null){
           jugadores.add(new Jugador(nombres.get(i)));
       }
    }
    
    private void inicializarTablero(){
        new Tablero();
    }
    
    private void salidaJugadores(){
        throw new UnsupportedOperationException("Sin Implementar");
    }
    
    //------------------------------------------------------------------------------------------------
    
    @Override
    public String toString() {
        //MODIFICAR MAS ADELANTE PARA MOSTRAR TODO EL JUEGO.
         return "{ \n Max Jugadores: " + MAX_JUGADORES + "\n Max Cartas: " + MAX_CARTAS + "\n Max. Casillas: " + MAX_CASILLAS + "\n Precio Libertad: " + PRECIO_LIBERTAD + "\n Saldo salida: " + SALDO_SALIDA +  " \n}";
    }
    
    
    
    
}

File: Practicas/Examen 1/QytetetRuby/pkg/QytetetRuby-0.0.1/lib/qytetet.rb
#encoding: utf-8

require "singleton"
require_relative "dado"

module ModeloQytetet
  
  class Qytetet
    
    include Singleton
    
    #ATRIBUTOS DE CLASE
    @@MAX_JUGADORES = 4
    @@MAX_CARTAS = 10
    @@MAX_CASILLAS = 20
    @@PRECIO_LIBERTAD = 200
    @@SALDO_SALIDA = 1000
    
        
    #ATRIBUTOS DE RELACIONES
    attr_accessor :cartaActual
    attr_accessor :jugadorActual
    attr_accessor :tablero
    
    dado = Dado.instance
    
    
    def initialize()
      @mazo = Array.new
      @cartaActual=nil
      @tablero = nil
      @jugadorActual = nil
      #inicializar_tablero
    end
    
    #--------------------------METODOS------------------------------
    
    def aplicar_sorpresa       
    end
    
    def cancelar_hipoteca      
    end
    
    def comprar_titulo_propiedad      
    end 
    
    def edificar_casa      
    end
    
    def edificar_hotel     
    end
    
    def hipotecar_propiedad(casilla)      
    end
    
    def inicializar_juego(nombres)      
    end
    
    def intentar_salir_carcel(metodo)      
    end
    
    def jugar      
    end
    
    def obtener_ranking(list)      
    end
    
    def propiedades_hipotecadas_jugador(hipotecadas)     
    end
    
    def siguiente_jugador     
    end
    
    def vender_propiedad(casilla)      
    end
    
    #METODOS PRIVADOS
    private
    def encarcelar_jugador
    end
    
    def inicializar_cartas_sorpresa
      @mazo<< Sorpresa.new("¡Felicidades! Hoy es dia de pago, recibes: ", 100, TipoSorpresa::PAGARCOBRAR)
      @mazo<< Sorpresa.new("¡Mala Suerte! Dia de impuestos, te toca pagar: ", 50, TipoSorpresa::PAGARCOBRAR)
      @mazo<< Sorpresa.new("¡Mala Suerte! Directo a la carcel", 5, TipoSorpresa::IRACASILLA)
      @mazo<< Sorpresa.new("El mazo de cartas te manda a la casilla numero ", 12, TipoSorpresa::IRACASILLA)
      @mazo<< Sorpresa.new("El mazo de cartas te manda a la casilla numero ", 19, TipoSorpresa::IRACASILLA)
      @mazo<< Sorpresa.new("Hacienda te cobra", 200, TipoSorpresa::PORCASAHOTEL)
      @mazo<< Sorpresa.new("Hacienda te devuelve", 50, TipoSorpresa::PORCASAHOTEL)
      @mazo<< Sorpresa.new("¡Felicidades!, es tu cumpleaños todos los jugadores te pagan ", 50, TipoSorpresa::PORJUGADOR)
      @mazo<< Sorpresa.new("Es tu dia de mala suerte, una apuesta te hace pagar a todos los jugadores", 50, TipoSorpresa::PORJUGADOR)
      @mazo<< Sorpresa.new("¡Enhorabuena!, quedas en libertad ", 0, TipoSorpresa::SALIRCARCEL)
    
      @mazo.shuffle! 
    end
    
    
    def inicializar_jugadores(nombres)
      nombres.each { |n|
        @jugadores << Jugador.new(n)
      }
    end
    
    
    def inicializar_tablero
      @tablero = Tablero.new
    end
    
    
    def salida_jugadores
      
    end
    
    public
    def to_s()
      "\n\tMAX Jugadores: #{@@MAX_JUGADORES} 
        MAX Cartas: #{@@MAX_CARTAS} 
        Precio Libertad: #{@@PRECIO_LIBERTAD} 
        Saldo Salida: #{@@SALDO_SALIDA}
      \n -TABLERO- #{@tablero}
      \n Mazo: #{@mazo}
      \n Jugador Actual: #{@jugadorActual}
      \n Carta Actual: #{@cartaActual}"
      
    end
    
    
  end #clase

end#modulo
